### PR TITLE
Change TR_ASSERT_FATAL to static_assert in TR_AliasSetInterface

### PR DESCRIPTION
**Pull request by ChungHsuanChen**
This patch elimates use of TR_ASSERT_FATAL to static assert in AliasSetInterface.hpp

Issue: [#4933](https://github.com/eclipse/omr/issues/4933)

### CODE REVIEW

File: compiler/il/AliasSetInterface.hpp [hunk 1]
```diff
@@ -24,6 +24,7 @@
 
 #include <stddef.h>
 #include <stdint.h>
+#include <cassert>
```
Comment (0xdaryl) [inline compiler/il/AliasSetInterface.hpp:27]: `static_assert` is a keyword and does not require an include

**Comment by 0xdaryl** [comment]
Also, please fix spelling in your commit body message.  Perhaps reword to:

Replace some uses of TR_ASSERT_FATAL with static_assert in AliasSetInterface.hpp

**Comment by ChungHsuanChen** [comment]
Sorry, I will request new PR

**Comment by ChungHsuanChen** [comment]
Commit need to clean

**Comment by ChungHsuanChen** [comment]
Close

**Pull request closed without merging**